test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/MSBuildAPIUtilityTests.cs

```diff
@@ -0,0 +1,385 @@
+                            loggers: null,
+                            toolsetDefinitionLocations: ToolsetDefinitionLocations.Default,
+                            // Having more than 1 node spins up multiple msbuild.exe instances to run builds in parallel
+                            // However, these targets complete so quickly that the added overhead makes it take longer
+                            maxNodeCount: 1,
+                            onlyLogCriticalEvents: false,
+                            loadProjectsReadOnly: false);
+
+            var projectOptions = new ProjectOptions
+            {
+                LoadSettings = ProjectLoadSettings.DoNotEvaluateElementsWithFalseCondition,
+                ProjectCollection = projectCollection
+            };
+
+            // Set up Directory.Packages.props file
+            var propsFile =
+@$"<Project>
+    <PropertyGroup>
+    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+    </PropertyGroup>
+    <ItemGroup>
+    <PackageVersion Include=""X"" Version=""1.0.0"" />
+    </ItemGroup>
+</Project>";
+            File.WriteAllText(Path.Combine(testDirectory, "Directory.Packages.props"), propsFile);
+
+            // Set up project file
+            string projectContent =
+@$"<Project Sdk=""Microsoft.NET.Sdk"">    
+	<PropertyGroup>                   
+	<TargetFramework>net6.0</TargetFramework>
+	</PropertyGroup>
+    <ItemGroup>
+    <PackageReference Include=""X"" />
+    </ItemGroup>
+</Project>";
+            File.WriteAllText(Path.Combine(testDirectory, "projectA.csproj"), projectContent);
+            var project = Project.FromFile(Path.Combine(testDirectory, "projectA.csproj"), projectOptions);
+
+            var msObject = new MSBuildAPIUtility(logger: new TestLogger());
+            // Get package version if it already exists in the props file. Returns null if there is no matching package version.
+            ProjectItem packageVersionInProps = project.Items.LastOrDefault(i => i.ItemType == "PackageVersion" && i.EvaluatedInclude.Equals("X"));
+
+            var libraryDependency = new LibraryDependency
+            {
+                LibraryRange = new LibraryRange(
+                        name: "X",
+                        versionRange: VersionRange.Parse("2.0.0"),
+                        typeConstraint: LibraryDependencyTarget.Package)
+            };
+
+            // Act
+            msObject.UpdatePackageVersion(project, packageVersionInProps, "2.0.0");
+
+            // Assert
+            Assert.Equal(projectContent, File.ReadAllText(Path.Combine(testDirectory, "projectA.csproj")));
+            Assert.Contains(@$"<PackageVersion Include=""X"" Version=""2.0.0"" />", File.ReadAllText(Path.Combine(testDirectory, "Directory.Packages.props")));
+            Assert.DoesNotContain(@$"<PackageVersion Include=""X"" Version=""1.0.0"" />", File.ReadAllText(Path.Combine(testDirectory, "Directory.Packages.props")));
```

Comment:
I think we can slightly improve the performance by reading text from the props file only once.

Reply:
Agreed, will fix that.